app/autodiscovery/src/main/java/com/fsck/k9/autodiscovery/thunderbird/ThunderbirdAutoconfigParser.kt

```diff
@@ -16,37 +16,32 @@ import org.xmlpull.v1.XmlPullParserFactory
  * [Autoconfig file format](https://wiki.mozilla.org/Thunderbird:Autoconfiguration:ConfigFileFormat)
  */
 class ThunderbirdAutoconfigParser {
-    fun parseSettings(stream: InputStream, email: String): ConnectionSettings? {
-        var incoming: ServerSettings? = null
-        var outgoing: ServerSettings? = null
-
+    fun parseSettings(stream: InputStream, email: String): DiscoveryResults? {
         val factory = XmlPullParserFactory.newInstance()
         val xpp = factory.newPullParser()
 
         xpp.setInput(InputStreamReader(stream))
 
+        val incomingServers = mutableListOf<DiscoveredServerSettings>()
```

Comment:
Not so keen on this very imperative implementation - it feels quite "noisy". Would much prefer a functional approach but I can revisit that near the end if feasible.